File: 001_217.ContainsDuplicate/solution.cpp
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    // Function to check if the input vector contains any duplicates
    bool containsDuplicate(vector<int>& v) {
        // Create an unordered_set to store unique elements
        unordered_set<int> s;
        
        // Get the size of the vector
        int n = v.size();

        // Iterate through each element in the vector
        for (int i = 0; i < n; i++) {
            
            // Check if the current element already exists in the set
            if (s.find(v[i]) != s.end()) {
                return true; // Duplicate found, return true
            }

            // Insert the element into the set
            s.insert(v[i]);
        }

        // If no duplicates found after the loop, return false
        return false;
    }
};

File: 001_217.ContainsDuplicate/solution.java
// Importing the HashSet class from java.util package
import java.util.HashSet;

class Solution {

    // Method to check if the array contains any duplicate elements
    public boolean containsDuplicate(int[] nums) {
        
        // Create a HashSet to store unique elements from the array
        HashSet<Integer> s = new HashSet<>();
        
        // Iterate through each element in the input array
        for (int num : nums) {
            // If the element is already in the set, it's a duplicate
            if (s.contains(num)) {
                return true; // Return true immediately if a duplicate is found
            }
            // Otherwise, add the element to the set
            s.add(num);
        }

        // If loop completes without finding duplicates, return false
        return false;
    }
}
